example docker-compose can't connect to the mysql DB

When using the docker-compose from the /exampes/ folder.
The connection to the DB can not be made.

`74) An exception was caught and reported. Message:
apollo_1  | Unable to obtain Jdbc connection from DataSource (jdbc:mysql://db:3306/apollo?createDatabaseIfNotExist=true) for user 'root': Could not create connection to database server.
apollo_1  | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
apollo_1  | SQL State  : 08001
apollo_1  | Error Code : 0
apollo_1  | Message    : Could not create connection to database server.
apollo_1  |
apollo_1  |   at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:138)
apollo_1  |
apollo_1  | 74 errors
apollo_1  | 	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:470)
apollo_1  | 	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:155)
apollo_1  | 	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107)
apollo_1  | 	at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:232)
apollo_1  | 	at com.netflix.governator.guice.LifecycleInjector.createChildInjector(LifecycleInjector.java:331)
apollo_1  | 	at com.netflix.governator.guice.LifecycleInjector.createInjector(LifecycleInjector.java:411)
apollo_1  | 	at com.netflix.governator.guice.LifecycleInjector.createInjector(LifecycleInjector.java:352)
apollo_1  | 	at io.logz.apollo.ApolloApplication.start(ApolloApplication.java:38)
apollo_1  | 	... 1 common frames omitted
apollo_1  | Caused by: org.flywaydb.core.internal.dbsupport.FlywaySqlException:
apollo_1  | Unable to obtain Jdbc connection from DataSource (jdbc:mysql://db:3306/apollo?createDatabaseIfNotExist=true) for user 'root': Could not create connection to database server.
apollo_1  | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
apollo_1  | SQL State  : 08001
apollo_1  | Error Code : 0
apollo_1  | Message    : Could not create connection to database server.
apollo_1  |
apollo_1  | 	at org.flywaydb.core.internal.util.jdbc.DriverDataSource.getConnectionFromDriver(DriverDataSource.java:419)
apollo_1  | 	at org.flywaydb.core.internal.util.jdbc.DriverDataSource.getConnection(DriverDataSource.java:381)
apollo_1  | 	at org.flywaydb.core.internal.util.jdbc.JdbcUtils.openConnection(JdbcUtils.java:51)
apollo_1  | 	at org.flywaydb.core.Flyway.execute(Flyway.java:1418)
apollo_1  | 	at org.flywaydb.core.Flyway.migrate(Flyway.java:971)
apollo_1  | 	at io.logz.apollo.di.ApolloMyBatisModule.migrateDatabase(ApolloMyBatisModule.java:81)
apollo_1  | 	at io.logz.apollo.di.ApolloMyBatisModule.initialize(ApolloMyBatisModule.java:40)
apollo_1  | 	at org.mybatis.guice.MyBatisModule.internalConfigure(MyBatisModule.java:134)
apollo_1  | 	at org.mybatis.guice.AbstractMyBatisModule.configure(AbstractMyBatisModule.java:66)
apollo_1  | 	at com.google.inject.AbstractModule.configure(AbstractModule.java:62)
apollo_1  | 	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)
apollo_1  | 	at com.google.inject.spi.Elements.getElements(Elements.java:110)
apollo_1  | 	at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:138)
apollo_1  | 	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:104)
apollo_1  | 	... 6 common frames omitted`

Pulled the new repo. New pull, new error :-(

`apollo_1  | 74) An exception was caught and reported. Message:
apollo_1  | Unable to obtain Jdbc connection from DataSource (jdbc:mysql://db.local:3306/apollo?createDatabaseIfNotExist=true) for user 'root': Communications link failure
apollo_1  |
apollo_1  | The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
apollo_1  | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
apollo_1  | SQL State  : 08S01
apollo_1  | Error Code : 0
apollo_1  | Message    : Communications link failure
apollo_1  |
apollo_1  | The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
apollo_1  |
apollo_1  |   at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:138)
apollo_1  |
apollo_1  | 74 errors
apollo_1  | 	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:470)
apollo_1  | 	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:155)
apollo_1  | 	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107)
apollo_1  | 	at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:232)
apollo_1  | 	at com.netflix.governator.guice.LifecycleInjector.createChildInjector(LifecycleInjector.java:331)
apollo_1  | 	at com.netflix.governator.guice.LifecycleInjector.createInjector(LifecycleInjector.java:411)
apollo_1  | 	at com.netflix.governator.guice.LifecycleInjector.createInjector(LifecycleInjector.java:352)
apollo_1  | 	at io.logz.apollo.ApolloApplication.start(ApolloApplication.java:38)
apollo_1  | 	... 1 common frames omitted
apollo_1  | Caused by: org.flywaydb.core.internal.dbsupport.FlywaySqlException:
apollo_1  | Unable to obtain Jdbc connection from DataSource (jdbc:mysql://db.local:3306/apollo?createDatabaseIfNotExist=true) for user 'root': Communications link failure`

--- a/examples/docker-compose.yml
+++ b/examples/docker-compose.yml
@@ -1,4 +1,4 @@
-version: '3'
+version: '2.1'
 
 services:
   db:
@@ -8,6 +8,10 @@ services:
     restart: always
     environment:
       MYSQL_ROOT_PASSWORD: rootpassword
+    volumes:
+      - ./mysql/docker-healthcheck:/usr/local/bin/docker-healthcheck
+    healthcheck:
+      test: "docker-healthcheck"
     networks:
       apollonet:
         aliases:
@@ -24,6 +28,9 @@ services:
     volumes:
       - ./apollo.conf:/root/apollo.conf
       - ./nginx.conf:/etc/nginx/nginx.conf
+    depends_on:
+      db:
+        condition: service_healthy
     networks:
       apollonet:
         aliases:

--- a/examples/mysql/docker-healthcheck
+++ b/examples/mysql/docker-healthcheck
@@ -1,0 +1,29 @@
+#!/bin/bash
+
+# Project: docker-library/healthcheck (https://github.com/docker-library/healthcheck)
+# File source: https://github.com/docker-library/healthcheck/blob/master/mysql/docker-healthcheck
+
+set -eo pipefail
+
+if [ "$MYSQL_RANDOM_ROOT_PASSWORD" ] && [ -z "$MYSQL_USER" ] && [ -z "$MYSQL_PASSWORD" ]; then
+	# there's no way we can guess what the random MySQL password was
+	echo >&2 'healthcheck error: cannot determine random root password (and MYSQL_USER and MYSQL_PASSWORD were not set)'
+	exit 0
+fi
+
+host="$(hostname --ip-address || echo '127.0.0.1')"
+user="${MYSQL_USER:-root}"
+export MYSQL_PWD="${MYSQL_PASSWORD:-$MYSQL_ROOT_PASSWORD}"
+
+args=(
+	# force mysql to not use the local "mysqld.sock" (test "external" connectibility)
+	-h"$host"
+	-u"$user"
+	--silent
+)
+
+if select="$(echo 'SELECT 1' | mysql "${args[@]}")" && [ "$select" = '1' ]; then
+	exit 0
+fi
+
+exit 1

